packages/frontend/src/features/sqlRunner/components/ContentPanel.tsx

```diff
@@ -230,59 +228,45 @@ export const ContentPanel: FC = () => {
                                 activeEditorTab === EditorTabs.VISUALIZATION
                             }
                         >
-                            {queryResults?.results && barChartConfig && (
-                                <SqlRunnerChart
-                                    data={queryResults}
-                                    config={barChartConfig}
-                                    isLoading={isLoading}
-                                    style={{
-                                        // NOTE: Ensures the chart is always full height
-                                        display:
-                                            selectedChartType ===
-                                            ChartKind.VERTICAL_BAR
-                                                ? 'block'
-                                                : 'none',
-                                        height: debouncedInputSectionHeight,
-                                        width: '100%',
-                                        flex: 1,
-                                    }}
-                                />
-                            )}
-
-                            {queryResults?.results && pieChartConfig && (
-                                <SqlRunnerChart
-                                    data={queryResults}
-                                    config={pieChartConfig}
-                                    isLoading={isLoading}
-                                    style={{
-                                        // NOTE: Ensures the chart is always full height
-                                        display:
-                                            selectedChartType === ChartKind.PIE
-                                                ? 'block'
-                                                : 'none',
-                                        height: debouncedInputSectionHeight,
-                                        width: '100%',
-                                        flex: 1,
-                                    }}
-                                />
-                            )}
-
-                            {queryResults?.results && (
-                                <Paper
-                                    shadow="none"
-                                    radius={0}
-                                    p="sm"
-                                    sx={() => ({
-                                        flex: 1,
-                                        overflow: 'auto',
-                                    })}
-                                >
-                                    <Table
-                                        data={queryResults.results}
-                                        config={tableVisConfig}
+                            {queryResults?.results &&
```

Comment:
For the future: 
We could rework this to take an array of visualizations and keep them mounted 

So at the moment when I switch between chart types (the echarts ones) I get them instantiated on every chart type change.

Reply:
Yea, good point, I saw this issue too. Let's treat it as a future improvement, but it's a good thought.